satpy/tests/reader_tests/test_fci_l1c_nc.py

```diff
@@ -282,236 +292,220 @@ def _get_reader_with_filehandlers(filenames, reader_configs):
+        """Test file pattern matching for TRAIL files, which should not be picked up."""
+        from satpy.readers import load_reader
 
         reader = load_reader(reader_configs)
         files = reader.select_files_from_pathnames(filenames)
-        # only 4 out of 5 above should match
-        assert len(files) == 4
+        assert len(files) == 0
 
-    _chans = {"solar": ["vis_04", "vis_05", "vis_06", "vis_08", "vis_09",
-                        "nir_13", "nir_16", "nir_22"],
-              "terran": ["ir_38", "wv_63", "wv_73", "ir_87", "ir_97", "ir_105",
-                         "ir_123", "ir_133"]}
+    @pytest.mark.parametrize('filehandler,channels,filenames,expected_res_n', [
+        (FakeFCIFileHandlerFDHSI, _chans_fdhsi, _test_filenames['fdhsi'], 16),
+        (FakeFCIFileHandlerHRFI, _chans_hrfi, _test_filenames['hrfi'], 4),
 
-    def test_load_counts(self, reader_configs):
+    ])
+    def test_load_counts(self, reader_configs, filehandler, channels, filenames,
+                         expected_res_n):
         """Test loading with counts."""
-        from satpy.tests.utils import make_dataid
-
-        # testing two filenames to test correctly combined
-        filenames = [
-            "W_XX-EUMETSAT-Darmstadt,IMG+SAT,MTI1+FCI-1C-RRAD-FDHSI-FD--"
-            "CHK-BODY--L2P-NC4E_C_EUMT_20170410114434_GTT_DEV_"
-            "20170410113925_20170410113934_N__C_0070_0067.nc",
-            "W_XX-EUMETSAT-Darmstadt,IMG+SAT,MTI1+FCI-1C-RRAD-FDHSI-FD--"
-            "CHK-BODY--L2P-NC4E_C_EUMT_20170410114442_GTT_DEV_"
-            "20170410113934_20170410113942_N__C_0070_0068.nc",
-        ]
-
-        reader = _get_reader_with_filehandlers(filenames, reader_configs)
-        res = reader.load(
-            [make_dataid(name=name, calibration="counts") for name in
-             self._chans["solar"] + self._chans["terran"]], pad_data=False)
-        assert 16 == len(res)
-        for ch in self._chans["solar"] + self._chans["terran"]:
-            assert res[ch].shape == (200 * 2, 11136)
-            assert res[ch].dtype == np.uint16
-            assert res[ch].attrs["calibration"] == "counts"
-            assert res[ch].attrs["units"] == "count"
-            if ch == 'ir_38':
-                numpy.testing.assert_array_equal(res[ch][~0], 1)
-                numpy.testing.assert_array_equal(res[ch][0], 5000)
-            else:
-                numpy.testing.assert_array_equal(res[ch], 1)
-
-    def test_load_radiance(self, reader_configs):
+        with mocked_basefilehandler(filehandler):
+            reader = _get_reader_with_filehandlers(filenames, reader_configs)
+            res = reader.load(
+                [make_dataid(name=name, calibration="counts") for name in
+                 channels["solar"] + channels["terran"]], pad_data=False)
+            assert expected_res_n == len(res)
+            for ch in channels["solar"] + channels["terran"]:
+                assert res[ch].shape == (200, 11136)
```

Comment:
Would this be 22272 for the solar HR channels?

Reply:
The current tests are assuming the same shape and geolocation parameters for both FDHSI and HRFI, since the computation is done exactly the same. But I agree that it would make the tests a bit nicer.. I'll look into it.

Reply:
Updated now to use realistic resolutions and shapes for each channel and data type input